x/vpool/types/pool.go

```diff
@@ -9,53 +9,61 @@ import (
 // HasEnoughQuoteReserve returns true if there is enough quote reserve based on
 // quoteReserve * tradeLimitRatio
 func (vpool *Vpool) HasEnoughQuoteReserve(quoteAmount sdk.Dec) bool {
-	return vpool.QuoteAssetReserve.Mul(vpool.Config.TradeLimitRatio).GTE(quoteAmount)
+	return vpool.QuoteAssetReserve.Mul(vpool.Config.TradeLimitRatio).GTE(quoteAmount.Abs())
```

Comment:
It seems that `Abs()` was intended, since the greater than condition would always be true for a negative `quoteAmount`